src/TurbulenceClosures/diffusivity_fields.jl

```diff
@@ -4,8 +4,11 @@ using Oceananigans.Fields: validate_field_tuple_grid
 ##### Forms for NonhydrostaticModel constructor
 #####
 
-DiffusivityFields(diffusivities::NamedTuple, grid, tracer_names, bcs, closure) =
+function DiffusivityFields(diffusivities::NamedTuple, grid, tracer_names, bcs, closure)
     validate_field_tuple_grid("diffusivities", diffusivities, grid)
+    
+    return diffusivities
```

Comment:
Does this have to be 

```julia
diffusivities = validate_field_tuple_grid("diffusivities", diffusivities, grid)
```

I haven't seen this `validate_field_tuple_grid` function before, it looks kinda sketchy... (I can't figure out what it does based on its name)

@simone-silvestri

Reply:
No that was the problem, like the other validation functions it just returns nothing if they're valid or errors if not.

As far as I can see this is the only instance where this function is used and the string is just used for the error message:
https://github.com/CliMA/Oceananigans.jl/blob/f3996c2703bb23b2e84f80c6f1564fc7965bdad1/src/Fields/field_tuples.jl#L119-L132

If it were to remain like that it probably would be a good idea to change the string to `diffusivity_fields` since that's what they're called in the model constructor.

Reply:
Yes, with that change I think we can merge this PR.

Some of the `validation` functions return a new copy of the validated objects, especially in grid constructors...